src/mlpack/methods/kernel_svm/kernel_svm_impl.hpp

```diff
@@ -0,0 +1,309 @@
+
+// In case it hasn't been included yet.
+#include "kernel_svm.hpp"
+
+namespace mlpack {
+namespace svm {
+
+template <typename MatType, typename KernelType>
+KernelSVM<MatType, KernelType>::KernelSVM(
+    const MatType& data,
+    const arma::Row<size_t>& labels,
+    const double regularization,
+    const bool fitKernel,
+    const size_t max_iter,
+    const double tol,
+    const KernelType kernel) :
+    regularization(regularization),
+    fitKernel(fitKernel),
+    kernel(kernel)
+{
+  intercept = 0;
+  Train(data, labels, max_iter, tol);
+}
+
+template <typename MatType, typename KernelType>
+KernelSVM<MatType, KernelType>::KernelSVM(
+    const size_t inputSize,
+    const double regularization,
+    const bool fitKernel,
+    const KernelType kernel) :
+    inputSize(inputSize),
+    regularization(regularization),
+    fitKernel(fitKernel),
+    kernel(kernel)
+{
+  intercept = 0;
+}
+
+template<typename MatType, typename KernelType>
+double KernelSVM<MatType, KernelType>::Train(
+    const MatType& data,
+    const arma::Row<size_t>& labels,
+    const size_t max_iter,
+    const double tol)
+{
+  std::cout<<"LOC 1"<<std::endl;
+  trainLabels = arma::Row<int> (data.n_cols);
+
+  // Changing labels values to 1, -1 values provided
+  // by user should 0 and 1.
+  std::cout<<"LOC 2"<<std::endl;
+  for (size_t i = 0; i < data.n_cols; i++)
+  {
+    if (labels(i) == 0)
+      trainLabels(i) = -1;
+    else
+      trainLabels(i) = 1;
+  }
+  std::cout<<"LOC 3"<<std::endl;
+
+  // Intializing variable to calculate alphas.
+  alpha = arma::zeros(data.n_cols);
+  size_t count = 0;
+
+  // Vector to store error value.
+  arma::vec E = arma::zeros(data.n_cols, 1);
+  // Eta value.
+  double eta = 0;
+
+  // Variable to store lower bound and higher bound.
+  double L = 0;
+  double H = 0;
+
+  // Storing kernel fucntion values.
+  arma::mat K = arma::mat(data.n_cols, data.n_cols);
+  std::cout<<"LOC 4"<<std::endl;
+
+  // Pre-compute the Kernel Matrix
+  for (size_t i = 0; i < data.n_cols; i++)
+  {
+    for (size_t j = 0; j < data.n_cols; j++)
+    {
+      K(i, j) = kernel.Evaluate(data.col(i), data.col(j));
+    }
+  }
+  std::cout<<"LOC 5"<<std::endl;
+
+  // Training starts from here.
+  while (count < max_iter)
+  {
+    size_t num_changed_alphas = 0;
+    for (size_t i = 0; i < data.n_cols; i++)
+    {
+      // Calculate Ei = f(x(i)) - y(i) using (2).
+      // E(i) = b + sum (X(i, :) * (repmat(alphas.*Y,1,n).*X)') - Y(i);
+      E(i) = intercept + arma::sum(alpha.t() % trainLabels % K.col(i).t())
+             - trainLabels(i);
+      std::cout<<"In Loop LOC 1"<<std::endl;
+      if ((trainLabels(i) * E(i) < -tol && alpha(i) < regularization) ||
+        (trainLabels(i) * E(i) > tol && alpha(i) > 0))
+      {
+        // In practice, there are many ways one can use to select
+        // the i and j. In this simplified code, we select them randomly.
+        size_t j = rand() % data.n_cols;
+        while (j == i || j >= data.n_cols)
```

Comment:
```suggestion
        while (j == i)
```
I think what you previously had before was fine. Also you could add assertions like
`assert(j < data.n_cols && j >= 0)` everywhere to check the range of the indices. If it is outside the range then the assertion would fail and throw an error. This would make it easier to debug.